sycl/include/sycl/handler.hpp

```diff
@@ -888,7 +906,11 @@ class __SYCL_EXPORT handler {
       extractArgsAndReqsFromLambda(reinterpret_cast<char *>(KernelPtr),
                                    KI::getNumParams(), &KI::getParamDesc(0),
                                    KI::isESIMD());
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+      MKernelName = detail::string(KI::getName());
+#else
```

Comment:
Why is this needed? Can't `detail::string` be copy-assigned directly from `KI::getName()`? If not, what overload for `operator=` could enable it? Can we add it instead?